integration_tests/src/main/python/array_test.py

```diff
@@ -394,3 +428,85 @@ def test_array_max_q1():
     def q1(spark):
         return spark.sql('SELECT ARRAY_MAX(TRANSFORM(ARRAY_REPEAT(STRUCT(1, 2), 0), s -> s.col2))')
     assert_gpu_and_cpu_are_equal_collect(q1)
+
+
+@pytest.mark.parametrize('data_gen', no_neg_zero_all_basic_gens + decimal_gens, ids=idfn)
+def test_array_intersect(data_gen):
+    if is_before_spark_313() and isinstance(data_gen, (FloatGen, DoubleGen)):
+        pytest.xfail("array_intersect tests on floating point types will fail because of NaN inequalilty in Spark 3.1.2 or earlier")
```

Comment:
I really dislike xfail tests because if they pass we do not fail the test. It would be much better to have a test that we know will fail because of the NaN issue and verify that we get back the error we expect, or the result that we expect.